dask/bytes/core.py

```diff
@@ -123,77 +121,97 @@ def read_bytes(urlpath, delimiter=None, not_zero=False, blocksize=2**27,
         raise IOError("%s resolved to no files" % urlpath)
     sizes = [fs.size(f) for f in names]
     out = []
+    # TODO: check fs for preferred locations of blocks here;
+    # and preferred blocksize?
+
     for size, name in zip(sizes, names):
-        blocksize = blocksize if blocksize is not None else size
-        offsets = list(range(0, size, blocksize))
+        bs = blocksize if blocksize is not None else size
+        offsets = list(range(0, size, bs))
         if len(offsets) > 1 and infer_compression(urlpath):
             raise ValueError('Cannot read compressed files (%s) in byte chunks,'
                              'use blocksize=None' % infer_compression(urlpath))
         if not_zero:
             offsets[0] = 1
         keys = ['read-block-%s-%s' % (offset, tokenize(name,
                 compression, offset, kwargs)) for offset in offsets]
-        # TODO: check fs for preferred locations of blocks here;
-        # and preferred blocksize?
-        func = lambda f, off: read_block(myopen(f, 'rb'), off, blocksize, delimiter)
+
+        def func(f, off):
+            with myopen(f, 'rb') as f:
```

Comment:
Recommend using `fn` or `filename` for filename pronoun to disambiguate with `f` for files.